rest-api-spec/src/main/resources/rest-api-spec/api/connector_sync_job.check_in.json

```diff
@@ -0,0 +1,35 @@
+{
+  "connector_sync_job.check_in": {
+    "documentation": {
+      "url": "https://www.elastic.co/guide/en/enterprise-search/current/connectors.html",
+      "description": "Checks in a connector sync job (refreshes 'last_seen')."
+    },
+    "stability": "experimental",
+    "visibility": "feature_flag",
+    "feature_flag": "es.connector_api_feature_flag_enabled",
+    "headers": {
+      "accept": [
+        "application/json"
+      ],
+      "content_type": [
```

Comment:
nit: since there is no `body` I don't think we need `content_type` header (`accept` seems to be still required)

Reply:
Adjusted with https://github.com/elastic/elasticsearch/pull/102842/commits/d571eeba9ae8d90b567e9506b1be1e5543c0380a